index.js

```diff
@@ -416,6 +431,32 @@ Primus.readable('forEach', function forEach(fn, done) {
 });
 
 /**
+ * Send a ping packet to all clients to ensure that they are still connected.
+ *
+ * @returns {Primus}
+ * @api private
+ */
+Primus.readable('heartbeat', function heartbeat() {
+  this.forEach(function forEach(spark) {
+    if (!spark.alive) {
```

Comment:
I agree that I don't like the complexity either. I'll explain my use case: I am thinking of legacy websocket clients that are not browsers (and thus won't understand Primus control messages) and don't implement protocol-level pings. They will be quite surprised when the connection is closed, even while traffic is flowing.

A majority of our consumers connect and submit no messages, and never had to (we made heartbeat opt-in by clearing each `spark.timeout` on init and on `'heartbeat'`). We'll need a backwards-compatible way to do that with Primus > 6.10.

For now, I can just monkey-patch Primus#heartbeat to recognize our own internal flags. That'll work fine without official support.